h3/src/frame/buf.rs

```diff
@@ -70,3 +70,57 @@ impl<T: Buf> Buf for BufList<T> {
         vecs
     }
 }
+
+pub struct Cursor<'a, B> {
+    buf: &'a B,
+    pos: usize,
+}
+
+impl<'a, B: Buf> Cursor<'a, B> {
+    pub fn new(buf: &'a B) -> Self {
+        Self { buf, pos: 0 }
+    }
+
+    pub fn position(&self) -> usize {
+        self.pos
+    }
+}
+
+impl<'a, B: Buf> Buf for Cursor<'a, B> {
+    #[inline]
+    fn remaining(&self) -> usize {
+        self.buf.remaining() - self.pos
+    }
+
+    #[inline]
+    fn bytes(&self) -> &[u8] {
+        &self.buf.bytes()[self.pos..]
```

Comment:
I think this won't work when used with a `BufList`, since calling `buf.bytes()` would only ever return the first buf's bytes.

Reply:
I've written a special `Cursor` for `BufList`, now it doesn't go in an infinite loop anymore.